src/Compilers/Core/Portable/Syntax/SyntaxTriviaList.cs

```diff
@@ -248,21 +249,31 @@ public SyntaxTriviaList InsertRange(int index, IEnumerable<SyntaxTrivia> trivia)
                 throw new ArgumentOutOfRangeException(nameof(index));
             }
 
-            var items = trivia.ToList();
-            if (items.Count == 0)
+            if (trivia == null)
             {
-                return this;
+                throw new ArgumentNullException(nameof(trivia));
             }
 
-            var list = this.ToList();
-            list.InsertRange(index, items);
+            var builder = SyntaxTriviaListBuilder.Create();
+
+            var current = 0;
+            foreach (var thisTrivia in this)
```

Comment:
Please rewrite using two ```for``` loops and get rid of the special handling for the case of adding to the end.